File: objc-2/h00.override/Triangle.m
#import "Triangle.h"

@implementation Triangle

  /******* INITIALIZERS *******/
  - (Triangle *) initWithWidth: (double) w height: (double) h {
    if (self = [super init]) { width = w; height = h; }
    return self;
  }

  /******* METHODS *******/
  - (double) calculateArea { return (width * height / 2); };

@end

File: javascripts/ts/h00.overriding/Triangle.ts
///<reference path='Shape.ts' />
/******* RHINO/SPIDERMONKEY SUPPORT *******/
load('Shape.js');
/******* CLASS DECLARATION *******/
class Triangle extends Shape {
  /******* CONSTRUCTORS *******/
  constructor(width : number, height : number) { super (width, height); }

  /******* METHODS *******/
  calculateArea() : number { return (this.width * this.height)/2; }
}
